src/wp-includes/class-wp-post-type.php

```diff
@@ -662,6 +662,23 @@ public function set_props( $args ) {
 	 */
 	public function add_supports() {
 		if ( ! empty( $this->supports ) ) {
+			/*
+			 * Move footnotes last.
+			 *
+			 * Footnotes are a special case and require other features to be supported:
+			 * 'editor', 'revisions' and 'custom-fields'. This moves the footnotes to
+			 * be added last so that the other features are already added before the
+			 * check in `add_post_type_support` is made.
+			 */
+			if ( isset( $this->supports['footnotes'] ) ) {
+				$footnotes = $this->supports['footnotes'];
+				unset( $this->supports['footnotes'] );
+				$this->supports['footnotes'] = $footnotes;
+			} elseif ( in_array( 'footnotes', $this->supports, true ) ) {
+				$index = array_search( 'footnotes', $this->supports, true );
+				unset( $this->supports[ $index ] );
+				$this->supports[] = 'footnotes';
+			}
```

Comment:
I'd suggest moving this code to its method and then invoking it here before the loop. What's going on here as a very specific purpose beyond but in support of `add_supports()`. It's like the prep work for before adding the supports.

Reply:
That said, +1 for the approach.